src/test/store/markers.test.js

```diff
@@ -583,3 +583,172 @@ describe('Marker schema filtering', function() {
+              null,
+              // One marker without cause
+              {
+                type: 'UserTiming',
+                name: 'name',
+                entryType: 'mark',
+              },
+              // One marker with a cause that was already processed in previous versions.
+              {
+                type: 'tracing',
+                category: 'RandomTracingMarker',
+                cause: {
+                  tid: 1111,
+                  time: 1001, // This was already processed
+                  stack: 0, // (root)
+                },
+              },
+              // One marker with a cause that wasn't already processed in
+              // previous versions.
+              {
+                type: 'FileIO',
+                source: 'PoisionOIInterposer',
+                filename: '/foo/bar/',
+                operation: 'create/open',
+                cause: {
+                  tid: 1111,
+                  time: 10, // This wasn't already processed
+                  stack: 0, // (root)
+                },
+              },
+            ],
+            length: 4,
+          },
+          stackTable: {
+            frame: [0], // (root)
+            prefix: [null],
+            category: [0],
+            subcategory: [0],
+            length: 1,
+          },
+          frameTable: {
+            address: [-1],
+            category: [0],
+            subcategory: [0],
+            func: [0], // (root)
+            innerWindowID: [null],
+            implementation: [null],
+            line: [null],
+            column: [null],
+            optimizations: [null],
+            length: 1,
+          },
+          stringArray: [
+            'no payload',
+            'UserTiming',
+            'RandomTracingMarker',
+            'FileIO',
+            '(root)',
+          ],
+          libs: [],
+          funcTable: {
+            address: [-1],
+            isJS: [true],
+            relevantForJS: [false],
+            name: [4], // (root)
+            resource: [-1],
+            fileName: [null],
+            lineNumber: [null],
+            columnNumber: [null],
+            length: 1,
+          },
+          resourceTable: {
+            lib: [],
+            name: [],
+            host: [],
+            type: [],
+            length: 0,
+          },
+        },
+      ],
+    };
+
+    const upgradedProfile = await unserializeProfileOfArbitraryFormat(profile);
+    const { getState } = storeWithProfile(upgradedProfile);
+
+    const getMarker = selectedThreadSelectors.getMarkerGetter(getState());
+    const derivedMarkers = selectedThreadSelectors
+      .getFullMarkerListIndexes(getState())
+      .map(getMarker);
+    // $FlowExpectError ignore Flow errors for simplicity.
+    expect(derivedMarkers[2].data.cause.time).toEqual(1001); // This hasn't changed, as expected.
+    // $FlowExpectError
+    expect(derivedMarkers[3].data.cause.time).toEqual(1010); // This changed, as expected.
```

Comment:
I wish we could use the optional chaining operator :/

Reply:
TBH I didn't think of it :D Does flow understand it?

Reply:
It's still experimental in our flow version and needs to be enabled explicitly :/